File: yh_tutorial/yh_tutorial_6/scripts/yh_server_6.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_6.srv import yh_srv_6, yh_srv_6Response

def srvCallback(req):
    return yh_srv_6Response(req.a + req.b + req.c)

def yh_server():
    rospy.init_node("yh_server_6")
    service_server = rospy.Service("yh_service_6", yh_srv_6, srvCallback)
    rospy.spin()

if __name__ == "__main__":
    yh_server()

File: yh_tutorial/yh_tutorial_6/src/yh_server_6.cpp
#include "ros/ros.h"
#include "yh_tutorial_6/yh_srv_6.h"

bool srvCallback(yh_tutorial_6::yh_srv_6::Request &req, yh_tutorial_6::yh_srv_6::Response &res)
{
    res.result = req.a + req.b + req.c;

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_server_6");
    ros::NodeHandle nh;

    ros::ServiceServer service_server = nh.advertiseService("yh_service_6", srvCallback);

    ROS_INFO("Service server ready.");

    ros::spin();

    return 0;
}


